projects/ngneat/helipopper/src/lib/tippy.directive.ts

```diff
@@ -136,6 +137,14 @@ export class TippyDirective implements OnChanges, AfterViewInit, OnDestroy, OnIn
     }
 
     this.setProps({ ...this.props, ...props });
+
+    if (this.checkContentChanges && isChanged<NgChanges<TippyDirective>>('content', changes)) {
+      if (!changes.content.previousValue && changes.content.currentValue) {
+        this.createInstance();
+      } else if (changes.content.previousValue && !changes.content.currentValue) {
+        this.destroyInstance();
+      }
+    }
```

Comment:
What `checkContentChanges` is flag used for? It looks like "isFirstChange"

Reply:
Without this this, createInstance from ngOnChanges is called before createInstance from ngAfterViewInit when content is not empty (createInstance is called twice). I want to use dynamic creating and destroying of instance only when createInstance was called from ngAfterViewInit to keep current functionality.

Reply:
1. I think you can rely on the `isFirstChange` property to skip the first update, or I would change the flag to `hasViewInit` and add an explanation to why.
2. Your implementation is making the lazy property irrelevant since an instance might be created before the element is in view.
3. Have you checked that tippy's `onHidden` is called when you destroy the instance? otherwise, some event listeners are leaking.

Reply:
1. changed to `isFirstChange.`
2. `ngAfterViewInit` body moved to `initInstanceCreation` function. It's called from `ngAfterViewInit` as well as from `ngOnChanges` (on content changed).
3. `onHidden` is called when instance is destroyed during tooltip is displayed. It's not called when instance is destroyed and tooltip is not displayed.